src/main/java/org/cyclops/integrateddynamics/core/evaluate/variable/ValueParseRegistry.java

```diff
@@ -0,0 +1,54 @@
+package org.cyclops.integrateddynamics.core.evaluate.variable;
+
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.tuple.Pair;
+import org.cyclops.integrateddynamics.api.evaluate.variable.IValue;
+import org.cyclops.integrateddynamics.api.evaluate.variable.IValueParseRegistry;
+import org.cyclops.integrateddynamics.api.evaluate.variable.IValueType;
+import org.cyclops.integrateddynamics.core.evaluate.operator.ParseOperator;
+import org.cyclops.integrateddynamics.core.evaluate.operator.Operators;
+
+import java.util.Map;
+
+/**
+ * Registry for parseing {@link IValue}.
+ * @author rubensworks/lostofthought
+ */
+public final class ValueParseRegistry implements IValueParseRegistry {
+
+    private static ValueParseRegistry INSTANCE = new ValueParseRegistry();
+
+    private final Map<IValueType, IMapping> mappings = Maps.newHashMap();
+
+    private ValueParseRegistry() {
+
+    }
+
+    /**
+     * @return The unique instance.
+     */
+    public static ValueParseRegistry getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public <T2 extends IValueType<V2>, V2 extends IValue> void register(T2 to, IMapping<T2, V2> mapping) {
+        mappings.put(to, mapping);
+        Operators.REGISTRY.register(new ParseOperator<>(to, mapping));
```

Comment:
I have yet to look into this in detail, but looking at this makes we wonder if this additional registry is even needed. As you are creating plain operators, is the operator registry not already sufficient then?

Reply:
This was honestly a hack on the cast operator.
When this was written, it was before I was made aware of 'name' and therefore was doing two way conversions.

Reply:
So I assume the value parse registry can be removed then?

Reply:
I would assume so. Trying to get more acquainted with the codebase as we speak.